Add async / await example to page object docs

antgel opened the pull request:
Changed old `getGreeting` to `getGreetingText` so that I could use `getGreeting` for the non-async example of returning the object. As requested https://github.com/angular/protractor/issues/4637.

antgel commented:
I signed it!

On 24 January 2018 at 19:40, googlebot <notifications@github.com> wrote:

> Thanks for your pull request. It looks like this may be your first
> contribution to a Google open source project. Before we can look at your
> pull request, you'll need to sign a Contributor License Agreement (CLA).
>
> 📝 *Please visit https://cla.developers.google.com/
> <https://cla.developers.google.com/> to sign.*
>
> Once you've signed, please reply here (e.g. I signed it!) and we'll
> verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address on your commit.
>    Check your existing CLA data <https://cla.developers.google.com/clas>
>    and verify that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If your company signed a CLA, they designated a Point of Contact who
>    decides which employees are authorized to participate. You may need to
>    contact the Point of Contact for your company and ask to be added to the
>    group of authorized contributors. If you don't know who your Point of
>    Contact is, direct the project maintainer to go/cla#troubleshoot. The email
>    used to register you as an authorized contributor must be the email used
>    for the Git commit.
>    - In order to pass this check, please resolve this problem and have
>    the pull request author add another comment and the bot will run again. If
>    the bot doesn't comment, it means it doesn't think anything has changed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/protractor/pull/4675#issuecomment-360214200>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AASCk3CyBg65TwDThC9cyiGeqdS7cGjZks5tN2sUgaJpZM4RrpVR>
> .
>



-- 
https:/linkedin.com/in/antgel
https://resume.antgel.org/
https://about.me/antonygelberg
https://antgel.org/

qiyigg commented:
Would you mind updating the comment according to https://github.com/angular/protractor/blob/master/CONTRIBUTING.md

Thanks a lot!

qiyigg commented:
Also, 
For now, we only have one paragraph to demonstrate how to write page object with asyns/await. But we don't have an example to show to use asyns/await page object in their test. Actually they still need await for the pageObject even though pageObject method itself have await, 
e.g.
await pageObject.getGreetingText(). That might be confusing since the sections after that are only for tests using control flow (don't have await for page object, don't have disable_control_flow in their test config) 

Maybe we can add more description or examples to show that?

antgel commented:
@qiyigg I updated the comment (I had read the contributing doc, but I wrongly assumed it wasn't relevant for docs), and added another example for `async / await` in the test itself.

qiyigg commented:
Merged. thanks!
